var/spack/repos/builtin/packages/detray/package.py

```diff
@@ -0,0 +1,83 @@
+# Copyright 2013-2024 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+
+class Detray(CMakePackage):
+    """Detray is a description library for high energy physics experiments that
+    works entirely without polymorphism, making it exceptionally suitable for
+    use on GPU platforms."""
+
+    homepage = "https://github.com/acts-project/detray"
+    url = "https://github.com/acts-project/detray/archive/refs/tags/v0.67.0.tar.gz"
+
+    tags = ["hep"]
+
+    maintainers("stephenswat")
+
+    license("MPL-2.0", checked_by="stephenswat")
+
+    version("0.69.0", sha256="d7de190c0c5af4567dc5cd30913be649b1652ebe8961e514484b99aaabd3051a")
+    version("0.68.0", sha256="6d57835f22ced9243fbcc29b84ea4c01878a46bfa5910e320c933e9bf8e96612")
+    version("0.67.0", sha256="87b1b29f333c955ea6160f9dda89628490d85a9e5186c2f35f57b322bbe27e18")
+
+    variant("csv", default=True, description="Enable the CSV IO plugin")
+    variant(
+        "cxxstd",
+        default="17",
+        values=("17", "20", "23"),
+        multi=False,
+        description="C++ standard used",
+    )
+    variant("json", default=True, description="Enable the JSON IO plugin")
+    variant(
+        "scalar",
+        default="float",
+        values=("float", "double"),
+        multi=False,
+        description="Scalar type to use by default",
+    )
+    variant("eigen", default=True, description="Enable the Eigen math plugin")
+    variant("smatrix", default=False, description="Enable the SMatrix math plugin")
+    variant("vc", default=True, description="Enable the Vc math plugin")
+
+    depends_on("cmake@3.11:", type="build")
+    depends_on("vecmem@1.6.0:")
+    depends_on("covfie@0.10.0:")
+    depends_on("nlohmann-json@3.11.0:", when="+json")
+    depends_on("dfelibs@20211029")
```

Comment:
Can this be open ended?
```suggestion
    depends_on("dfelibs@20211029:")
```

Reply:
@stephenswat If it works to open-end the version range, it may be good.

But of course, if only this version is guaranteed to work then of course, you've to reject this suggestion.

In any case: If you update the version, and this needs to be pinned to this version, you might want to add a comment in case the question comes up again in the future.